cmd/snap/cmd_run.go

```diff
@@ -944,6 +945,18 @@ func (x *cmdRun) runSnapConfine(info *snap.Info, securityTag, snapApp, hook stri
 	}
 	env := snapenv.ExecEnv(info, extraEnv)
 
+	// Systemd automatically places services under a unique cgroup encoding the
+	// security tag, for apps and hooks we need to create a transient scope
+	// with similar purpose.
```

Comment:
Please flesh this out more. It took us a long time to get to this point, so documenting how/why this all works together, with links to things like https://www.freedesktop.org/wiki/Software/systemd/ControlGroupInterface/ will make this seem less magical after we move on to other things and come back to the code in a year.